### ISSUE TITLE
[experimental plotting] 2-D plot without coord raises

### ISSUE THREAD

**Issue by SimonHeybrock** [scipp/scipp#2810]
```python
import scipp as sc
sc.plotting.select_backend('experimental')

da = sc.data.table_xyz(100).bin(x=10, y=12).bins.mean()
del da.coords['x']
da.plot()
```
raises
```python
TypeError: Dimensions of C (10, 12) are incompatible with X (13) and/or Y (10); see help(pcolormesh)
```

**Comment by nvaytet** [comment]
Hmm i think this is partly solved in the `kwargs_for_new_plotting` branch (by adding the `shading='auto'` when calling `pcolormesh`), but I am still getting a warning
```
/home/nvaytet/code/scipp/jupyter/scipp/experimental/plotting/mesh.py:66: MatplotlibDeprecationWarning: shading='flat' when X and Y have the same dimensions as C is deprecated since 3.3.  Either specify the corners of the quadrilaterals with X and Y, or pass shading='auto', 'nearest' or 'gouraud', or set rcParams['pcolor.shading'].  This will become an error two minor releases later.
  self._mesh = self._ax.pcolormesh(self._data.meta[self._dims['x']].values,
```
which I don't understand.

In addition, I think that when coords are missing, the wrapper `plot` function actually adds the fake coords onto the original data, which is BAD.
(if you print `da` again after plotting it will have an `x` coord)

**Comment by SimonHeybrock** [comment]
Oddly it works when I also `del da.coords['y']`.

**Comment by nvaytet** [comment]
It might be because the `y` coord is a bin-edge

**Comment by nvaytet** [comment]
Fixed in #2796
